lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -380,6 +386,42 @@ class ReportUIFeatures {
     this._document.body.removeChild(a);
     setTimeout(_ => URL.revokeObjectURL(href), 500);
   }
+
+  _handleStickyHeader() {
+    const topbarEl = this._document.querySelector('.lh-topbar');
+    const scoreScaleEl = this._document.querySelector('.lh-scorescale');
+    const stickyHeaderEl = this._document.querySelector('.lh-sticky-header');
+    const highlightEl = this._document.querySelector('.lh-highlighter');
+    if (!topbarEl || !scoreScaleEl || !stickyHeaderEl || !highlightEl) return;
+
+    // Show sticky header when the score scale begins to go underneath the topbar.
+    const showStickyHeader =
+      topbarEl.getBoundingClientRect().bottom - scoreScaleEl.getBoundingClientRect().top >= 0;
+    stickyHeaderEl.classList.toggle('lh-sticky-header--stuck', showStickyHeader);
+
+    // Highlight mini gauge when section is in view.
+    // Use the middle of the viewport as an anchor - the closest category to the middle
+    // is the one "in view".
+    let highlightIndex = 0;
+    let highlightIndexDistance = Number.POSITIVE_INFINITY;
+    const categoryEls = this._document.querySelectorAll('.lh-category');
+    for (const [index, categoryEl] of Object.entries(categoryEls)) {
+      // Normalize to middle of viewport.
+      const distanceToMiddle = categoryEl.getBoundingClientRect().top - window.innerHeight / 2;
+      // Closest negative distance to zero wins.
+      if (distanceToMiddle < 0 && highlightIndexDistance > distanceToMiddle) {
```

Comment:
seems like we could simplify some of this but maybe I'm misunderstanding

- `highlightIndexDistance` will only ever update when `distanceToMiddle` is negative
- `highlightIndexDistance` is `-distanceToMiddle`, so `highlightIndexDistance` is always positive
-  `highlightIndexDistance` will always be greater than `distanceToMiddle` if `distanceToMiddle` is negative, so it's really just `if (distanceToMiddle < 0)`

this seems equivalent to "we highlight the last category that starts above the middle of the window" which at least for me is a lot easier to understand.

maybe something like the below?
```js
const categoriesAboveTheMiddle = categories.filter(el => el.getBoundingClientRect().top - window.innerHeight / 2)
const categoryToHighlight = categoriesAboveTheMiddle[categoriesAboveTheMiddle.length - 1]
const guageToHighlight = gauges[categories.indexOf(categoryToHighlight)]

Reply:
this works:

```js
// Highlight mini gauge when section is in view.
// In view = the last category that starts above the middle of the window.
const categoryEls = Array.from(this._document.querySelectorAll('.lh-category'));
const categoriesAboveTheMiddle =
  categoryEls.filter(el => el.getBoundingClientRect().top - window.innerHeight / 2 < 0);
const highlightIndex =
  categoriesAboveTheMiddle.length > 0 ? categoriesAboveTheMiddle.length - 1 : 0;

// Category order matches gauge order in sticky header.
// TODO(hoten): not 100% true yet, need to order gauges like: core, pwa, plugins. Remove
// this comment when that is done.
const gaugeToHighlight = stickyHeaderEl.querySelectorAll('.lh-gauge__wrapper')[highlightIndex];
// @ts-ignore
highlightEl.style.left = gaugeToHighlight.getBoundingClientRect().left + 'px';
```

thanks for making more declarative, i'll update